Do not store undefined values in store

mhajas opened the pull request:
Closes #10744

This PR changes the new store to not include undefined values in the store. For example, if we have a field with the type `Map<String, String>` any key that is mapping to a null value won't be included in the store. The same is true also for Collections.

[MapRealmEntityImpl](https://gist.github.com/mhajas/c9e9abdd83823d6e9871114560c57e40)
[MapComponentEntityImpl](https://gist.github.com/mhajas/a6d10f7c0b0e20f75408abe7e5c26e49)

mhajas committed:
Do not store undefined values in store

Closes #10744

hmlnarik commented on model/build-processor/src/main/java/org/keycloak/models/map/processor/AbstractGenerateEntityImplementationsProcessor.java:
```diff
@@ -215,6 +215,25 @@ protected String deepClone(TypeMirror fieldType, String parameterName) {
+
+    protected String isUndefined(TypeMirror fieldType, String parameterName) {
+        TypeElement typeElement = elements.getTypeElement(types.erasure(fieldType).toString());
+        return parameterName  + " == null" + (
+                isCollection(fieldType)
+                ? " || " + parameterName + ".isEmpty()" + (
+                        !isMapType(typeElement)
+                        ? " || " + parameterName + ".stream().allMatch(e -> " + isUndefined(getGenericsDeclaration(fieldType).get(0), "e") + ")"
+                        : "")
+                : "");
```
This should be optimized to leverage `Objects::isNull` reference rather than creating many `(e -> e == null)` functions when the field type is not a collection.

mhajas replied:
It would be possible. We can even do it similarly as we do it within HotRod. We can have some util class with overloaded `isUndefined` method. This way we could even specify class specific undefined conditions e.g. `"".equals(stringParameter.trim())`.

mhajas commented:
@hmlnarik Thanks for the review! I updated the PR based on your comment. I also updated Github gists in the first comment to reflect the newest version.

hmlnarik commented on model/map/src/main/java/org/keycloak/models/map/common/UndefinedValuesUtils.java:
```diff
@@ -0,0 +1,50 @@
+import java.util.Collection;
+import java.util.Map;
+
+public class UndefinedValuesUtils {
```
Given that this is a class in map.common package, this class and its methods deserve javadoc
